src/test/java/com/hillel/items_exchange/controller/LocationControllerIntegrationTest.java

```diff
@@ -182,4 +190,28 @@ void deleteLocations_shouldDeleteExistedLocation() throws Exception {
                 .andExpect(status().isOk())
                 .andReturn();
     }
-}
\ No newline at end of file
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DataSet("database_init.yml")
+    void createLocationsInitFile_whenDataIsValid_shouldCreateFileAndReturnItsContent() throws Exception {
+        MvcResult response = mockMvc.perform(post("/location/locations-init")
+                .param("data", Files.readString(pathToFileParseLocationsFrom)))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andReturn();
+        assertTrue(response.getResponse().getContentAsString().contains("2580"));
```

Comment:
Instead of hard-coding the value here you could use **locationRepository** and it's method **count** for checking equality.